src/components/sbb-file-selector/sbb-file-selector.tsx

```diff
@@ -0,0 +1,296 @@
+  tag: 'sbb-file-selector',
+})
+export class SbbFileSelector implements ComponentInterface {
+  /** Whether the component has a dropzone area or not. */
+  @Prop() public variant: InterfaceSbbFileSelectorAttributes['variant'] = 'default';
+
+  /** Whether more than one file can be selected. */
+  @Prop() public multiple: boolean;
+
+  /** Whether the newly added files should override the previously added ones. */
+  @Prop() public multipleMode: InterfaceSbbFileSelectorAttributes['multipleMode'];
+
+  /** A comma-separated list of allowed unique file type specifiers. */
+  @Prop() public accept: string;
+
+  /** The title displayed in `dropzone` variant. */
+  @Prop() public titleContent?: string;
+
+  /** Whether the component is disabled. */
+  @Prop() public disabled: boolean;
+
+  /** The list of selected files. */
+  @State() public files: File[];
+
+  /** Current document language used for translations. */
+  @State() private _currentLanguage = documentLanguage();
+
+  /**
+   * State of listed named slots, by indicating whether any element for a named slot is defined.
+   */
+  @State() private _namedSlots = createNamedSlotState('error');
+
+  /** An event which is emitted each time the file list changes. */
+  @Event({
+    eventName: 'file-changed',
+  })
+  public fileChangedEvent: EventEmitter<File[]>;
+
+  /** An event emitted on error. */
+  @Event({ bubbles: true }) public error: EventEmitter;
+
+  @Element() private _element!: HTMLElement;
+
+  private _loadButton: HTMLElement;
+  private _dragTarget: HTMLElement;
+  private _suffixes: string[] = ['B', 'kB', 'MB', 'GB', 'TB'];
+
+  private _handlerRepository = new HandlerRepository(
+    this._element,
+    languageChangeHandlerAspect((l) => (this._currentLanguage = l)),
+    namedSlotChangeHandlerAspect((m) => (this._namedSlots = m(this._namedSlots))),
+  );
+
+  public connectedCallback(): void {
+    this._handlerRepository.connect();
+  }
+
+  public disconnectedCallback(): void {
+    this._handlerRepository.disconnect();
+  }
+
+  private _blockEvent(event): void {
+    event.stopPropagation();
+    event.preventDefault();
+  }
+
+  private _checkFileEquality(file1: File, file2: File): boolean {
+    return (
+      file1.name === file2.name &&
+      file1.size === file2.size &&
+      file1.lastModified === file2.lastModified
+    );
+  }
+
+  private _onDragEnter(event): void {
+    if (!this.disabled) {
+      this._setDragState(event.target, true);
+      this._blockEvent(event);
+    }
+  }
+
+  private _onDragLeave(event): void {
+    if (!this.disabled && event.target === this._dragTarget) {
+      this._setDragState();
+      this._blockEvent(event);
+    }
+  }
+
+  private _onFileDrop(event): void {
+    if (!this.disabled) {
+      this._setDragState();
+      this._blockEvent(event);
+      try {
+        const files = this.multiple ? event.dataTransfer.files : [event.dataTransfer.files[0]];
+        this._createFilesList(files);
+      } catch (e) {
```

Comment:
which error can happen here that it ever enters the catch block?

Reply:
TBH i have no idea, except maybe selecting a file and then deleting it via file system

Reply:
But even then, it's already in browser cache

Reply:
If there is no error available, we can remove our logic for it
I was under the impression that dragging and dropping files could result in an error, but I could be wrong

Reply:
Removed in [85ed886](https://github.com/lyne-design-system/lyne-components/pull/1894/commits/85ed8862c66d77bbe3adc4c3b942fdbf4d85dfe0)